src/common/utils/storage.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+
+// a single place to define all used localStorage keys, to avoid collisions or hardcoded strings
+export enum StorageKeys {
+  COUNTY_MAP_TYPE = 'COUNTY_MAP_TYPE',
+  FULLSTORY_RECORD = 'FULLSTORY_RECORD',
+}
+
+// From https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#testing_for_availability
+export function storageAvailable(storageType: string) {
+  let storage;
+  const testKey = '__storage_test__';
+  try {
+    // @ts-ignore
+    storage = window[storageType];
+    storage.setItem(testKey, testKey);
+    storage.removeItem(testKey);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+// Hook from https://usehooks.com/useLocalStorage/
+// but updated to fallback to setState if localstorage is not available.
+export function useLocalStorage<T>(
+  key: StorageKeys,
+  initialValue: T,
+): [T, (value: T) => void] {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
```

Comment:
to simplify maybe:
```
let item
try {
  if (storageAvailable('localStorage')) {
    item = window.localStorage.getItem(key);
    item = JSON.parse(item)
  }
} catch {
  item = initialValue as T
}

return item
```

basically, just to put the JSON parsing closer to the `getItem` call to make it clearer that those actions go hand-in-hand.

Reply:
had to tweak this slightly because item is nullable.